bundles/core/src/main/java/com/adobe/cq/email/core/components/util/StyleTokenFactory.java

```diff
@@ -137,19 +145,35 @@ public static void addProperties(StyleToken styleToken, String properties) {
     @Nullable
     public static String toCss(StyleToken styleToken) {
         String css = null;
-        if (Objects.nonNull(styleToken)) {
-            css = styleToken.getSelector() + " { " + getAllProperties(styleToken) + " }\n";
+        StringBuilder cssBuilder = toCss(styleToken, new StringBuilder());
+        if (StringUtils.isNotEmpty(cssBuilder.toString())) {
```

Comment:
Call `cssBuilder.toString()` only once as this will create a copy of the `char[]` the StringBuilder uses each time.

Reply:
fixed in d499c2a0f25f74133accb7f777c37037b5764dad